addon/mixins/route.js

```diff
@@ -168,11 +168,13 @@ export default Ember.Mixin.create({
         this.set('_currentPage', startingPage);
         this.set('_totalPages', totalPages);
         infinityModel.set('reachedInfinity', !this.get('_canLoadMore'));
-        Ember.run.scheduleOnce('afterRender', this, 'infinityModelUpdated', {
-          lastPageLoaded: startingPage,
-          totalPages: totalPages,
-          newObjects: infinityModel
-        });
+        if(this.infinityModelUpdated) {
```

Comment:
I ran into occassions, where the infinityModelUpdated and infinityModelLoaded hooks weren't defined, so ember was complaining about running a callback with an undefined method.

Maybe before ember 2.0.0, thse things we're possible or these issues are independent of the update.